plugins/vsphere/pkg/command/source/delete.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package source
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/vmware-tanzu/sources-for-knative/plugins/vsphere/pkg"
+)
+
+func NewSourceDeleteCommand(clients *pkg.Clients, opts *Options) *cobra.Command {
+	result := cobra.Command{
+		Use:   "delete",
+		Short: "Delete a vSphere source",
+		Long:  "Delete a vSphere source",
+		Example: `# Delete the source in the default namespace
+kn vsphere source delete --name vc-01-source
+
+# Delete the source in the specified namespace
+kn vsphere source delete --namespace ns --name vc-01-source
+`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Name == "" {
+				return fmt.Errorf("'name' requires a nonempty name provided with the --name option")
+			}
+			return nil
+		},
```

Comment:
Again, consider using positional args

Reply:
Agree, I will open an issue for this as it would be a bigger rewrite which I'd like to defer for now to get this in.

Reply:
https://github.com/vmware-tanzu/sources-for-knative/issues/347